src/HTTP.jl

```diff
@@ -456,16 +460,18 @@ hard-codes the value of the `retry` and `redirect` keyword arguments. When we pa
 effectively over-writing the default and any user-provided keyword arguments for `retry` or `redirect`.
 """
 macro client(requestlayers, streamlayers=[])
-    esc(quote
-        get(a...; kw...) = request("GET", a...; kw...)
-        put(a...; kw...) = request("PUT", a...; kw...)
-        post(a...; kw...) = request("POST", a...; kw...)
-        patch(a...; kw...) = request("PATCH", a...; kw...)
-        head(a...; kw...) = request("HEAD", a...; kw...)
-        delete(a...; kw...) = request("DELETE", a...; kw...)
-        open(f, a...; kw...) = request(a...; iofunction=f, kw...)
-        request(method, url, h=HTTP.Header[], b=HTTP.nobody; headers=h, body=b, query=nothing, kw...)::HTTP.Response =
+    return @remove_linenums! esc(quote
+        get(a...; kw...) = ($__source__; request("GET", a...; kw...))
+        put(a...; kw...) = ($__source__; request("PUT", a...; kw...))
+        post(a...; kw...) = ($__source__; request("POST", a...; kw...))
+        patch(a...; kw...) = ($__source__; request("PATCH", a...; kw...))
+        head(a...; kw...) = ($__source__; request("HEAD", a...; kw...))
+        delete(a...; kw...) = ($__source__; request("DELETE", a...; kw...))
+        open(f, a...; kw...) = ($__source__; request(a...; iofunction=f, kw...))
+        function request(method, url, h=HTTP.Header[], b=HTTP.nobody; headers=h, body=b, query=nothing, kw...)::HTTP.Response
+            $__source__
             HTTP.request(HTTP.stack($requestlayers, $streamlayers), method, url, headers, body, query; kw...)
+        end
     end)
```

Comment:
> but 👍 i still think your proposal in https://github.com/JuliaLang/julia/pull/31335 is sound

👍 But i actually kind of really like the solution you have taken in this PR. Maybe we should just move your macro into Base, and call it `@remove_macro_linenums!` or `@remove_new_linenums!` or something, since it only removes the linenums from the local quote expression but leaves any that are part of the data of the expression, which is actually a really nice combination.

The more I think about it, I actually do think that the macro version you have is really _exactly_ what you want! It says: remove the linenums added by the fact that I'm creating an expression here, but don't remove any of the linenums that are in the data of my expression, either because they were passed-in by the caller, or because i've added them via `$__source__`. That's exactly what you want!

So i think that your solution is maybe _even better_ and we should just push yours up into Base? What do you think?